crates/sui-framework/packages/sui-framework/sources/multiparty.move

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::multiparty;
+
+use sui::vec_map::{Self, VecMap};
+
+/// A party can read the object, taking it as an immutable argument. This restriction is checked
+/// when sending the transaction.
+const READ: u8 = 0x01;
+
+/// The party can mutate the object, but not change its owner or delete it. This is checked at
+/// end end of transaction execution.
+const WRITE: u8 = 0x02;
+
+/// The party can delete the object, but not otherwise modify it. This is checked at the end of
+/// transaction execution.
+const DELETE: u8 = 0x04;
+
+/// The party can change the owner of the object, but not otherwise modify it. This is checked at
+/// the end of transaction execution.
+const TRANSFER: u8 = 0x08;
+
+/// No permissions.
+const NO_PERMISSIONS: u64 = 0;
+
+/// Permissions corresponding to shared objects as defined by `sui::transfer::share_object`.
+const LEGACY_SHARED: u64 = (READ | WRITE | DELETE) as u64;
```

Comment:
sorry if I'm late to this discussion - I think we don't want to have any "legacy shared" stuff in here initially?